src/main/java/webserver/ResponseEntity.java

```diff
@@ -0,0 +1,185 @@
+package webserver;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class ResponseEntity {
+	private final String protocol;
+	private final int statusCode;
+	private final String reasonPhrase;
+	private final String contentType;
+	private final int contentLength;
+	private final String location;
+	private final Map<String, String> cookies;
+	private final byte[] body;
+
+	private ResponseEntity(String protocol, int statusCode, String reasonPhrase, String contentType, int contentLength, String location, Map<String, String> cookies, byte[] body) {
+		this.protocol = protocol;
+		this.statusCode = statusCode;
+		this.reasonPhrase = reasonPhrase;
+		this.contentType = contentType;
+		this.contentLength = contentLength;
+		this.location = location;
+		this.cookies = cookies;
+		this.body = body;
+	}
+
+	public static ContentType response200() {
+		return builder()
+			.protocol("HTTP/1.1")
+			.statusCode(200)
+			.reasonPhrase("OK");
+	}
+
+	public static HttpResponseHeader response200Html() {
+		return builder()
+			.protocol("HTTP/1.1")
+			.statusCode(200)
+			.reasonPhrase("OK")
+			.contentType("text/html;charset=utf-8");
+	}
+
+	public static HttpResponseHeader response302Html() {
+		return builder()
+			.protocol("HTTP/1.1")
+			.statusCode(302)
+			.reasonPhrase("FOUND")
+			.contentType("text/html;charset=utf-8");
+	}
+
+	public void response(DataOutputStream dataOutputStream) throws IOException {
+		dataOutputStream.writeBytes(protocol + " " + statusCode + " " + reasonPhrase + " \r\n");
+		dataOutputStream.writeBytes("Content-Type: " + contentType + "\r\n");
+		dataOutputStream.writeBytes("Content-Length: " + contentLength + "\r\n");
+
+		if (Objects.nonNull(location) && !location.isBlank()) {
+			dataOutputStream.writeBytes("Location: " + location + "\r\n");
+		}
+
+		setCookiesResponseHeader(dataOutputStream);
+		dataOutputStream.writeBytes("\r\n");
+
+		dataOutputStream.write(body, 0, body.length);
+		dataOutputStream.flush();
+	}
+
+	private void setCookiesResponseHeader(DataOutputStream dataOutputStream) throws IOException {
+		Set<String> cookieKeys = cookies.keySet();
+
+		if (!cookieKeys.isEmpty()) {
+			dataOutputStream.writeBytes("Set-Cookie: ");
+
+			for (String key : cookieKeys) {
+				dataOutputStream.writeBytes(key + "=" + cookies.get(key) + "; ");
+			}
+
+			dataOutputStream.writeBytes("path=/;");
+			dataOutputStream.writeBytes("\r\n");
+		}
```

Comment:
지금도 좋지만 Depth가 2여서
```suggestion
		StringBuilder sb = new StringBuilder();
		
		for(Map.Entry<String, String> entrySet : cookies.entrySet()) {
			sb.append(entrySet.getKey() + "=" + entrySet.getValue() + ";");
		}
		
		if (sb.isEmpty()) {
			return;
		}
		
		Set<String> cookieKeys = cookies.keySet();
		dataOutputStream.writeBytes("Set-Cookie: ");
		dataOutputStream.writeBytes(sb.toString());
		dataOutputStream.writeBytes("path=/;");
		dataOutputStream.writeBytes("\r\n");
```

요렇게 작성하면 더 깔끔할 것 같아요